### PR TITLE
Add paragraph with descheduler annotation

### PR DESCRIPTION
**Pull request by ksimon1**
Add paragraph with descheduler annotation
Signed-off-by: Karel Simon <ksimon@redhat.com>

### CODE REVIEW

File: usage/node-placement.md [hunk 1]
```diff
@@ -143,3 +143,16 @@ An example for `tolerations` may look like this:
...
+----------------------
+In some cases we might need to rebalance the cluster on current scheduling policy 
+and load conditions. [Descheduler](https://github.com/kubernetes-sigs/descheduler) 
+can find pods, which violates e.g. scheduling decisions and evict them based on descheduler 
+policies. Kubevirt VMs are handled as pods with local storage, so by default, 
+descheduler will not evict them. But it can be easily overridden by adding special annotation to vm:
+
+      descheduler.alpha.kubernetes.io/evict: true
+
+This annotation will case, that descheduler will be able to evict VM's pod which can be then
```
Comment (codificat) [inline usage/node-placement.md:157]: Typo (and some grammar suggestions):

```suggestion
This annotation will cause that the de-scheduler will be able to evict the VM's pod which can then be 
```

File: usage/node-placement.md [hunk 2]
```diff
@@ -143,3 +143,16 @@ An example for `tolerations` may look like this:
...
+In some cases we might need to rebalance the cluster on current scheduling policy 
+and load conditions. [Descheduler](https://github.com/kubernetes-sigs/descheduler) 
+can find pods, which violates e.g. scheduling decisions and evict them based on descheduler 
+policies. Kubevirt VMs are handled as pods with local storage, so by default, 
+descheduler will not evict them. But it can be easily overridden by adding special annotation to vm:
+
+      descheduler.alpha.kubernetes.io/evict: true
+
+This annotation will case, that descheduler will be able to evict VM's pod which can be then 
+scheduled by scheduler on different nodes. Users should know how and if the pod will be recreated.
```
Comment (codificat) [inline usage/node-placement.md:158]: Question: how will users know how/if the pod will be recreated? Maybe add a reference here?

**Review by codificat — CHANGES_REQUESTED** [review]
Just a minor suggestion and a question

**Comment by mazzystr** [comment]
@ksimon1 We have bad timing on your contribution.  The user-guide migration to mkdocs markdown engine and restructuring the page hierarchy completed today.

Your pull req now needs a rebase.  The rebase will be difficult. `usage/node-placement.md` has been renamed to `docs/operations/node_assignment.md`.  You will need to integrate this diff into the new file.  Another option is pull master, recreate your branch, recreate the content in the new file, create new pull req.  Yet another option is for me to do it.   The con to that is you don't get the GitHub credit.

I'm happy to discuss further in [slack](https://kubernetes.slack.com/archives/C8ED7RKFE)

File: docs/operations/node_assignment.md [hunk 3]
```diff
@@ -140,3 +140,17 @@ An example for `tolerations` may look like this:
         operator: "Equal"
         value: "value"
         effect: "NoSchedule"
+
+## Node balancing with Descheduler
+
+In some cases we might need to rebalance the cluster on current scheduling policy 
+and load conditions. [Descheduler](https://github.com/kubernetes-sigs/descheduler) 
+can find pods, which violates e.g. scheduling decisions and evict them based on descheduler 
+policies. Kubevirt VMs are handled as pods with local storage, so by default,
```
Comment (rmohr) [inline docs/operations/node_assignment.md:149]: It needs to be but on the VMI template on the VM:

Something like:

```
spec:
  template:
    metadata:
      labels:
        descheduler.alpha.kubernetes.io/evict: true
```

File: docs/operations/node_assignment.md [hunk 4]
```diff
@@ -140,3 +140,17 @@ An example for `tolerations` may look like this:
...
+
+## Node balancing with Descheduler
+
+In some cases we might need to rebalance the cluster on current scheduling policy 
+and load conditions. [Descheduler](https://github.com/kubernetes-sigs/descheduler) 
+can find pods, which violates e.g. scheduling decisions and evict them based on descheduler 
+policies. Kubevirt VMs are handled as pods with local storage, so by default, 
+descheduler will not evict them. But it can be easily overridden by adding special annotation to vm:
+
+    descheduler.alpha.kubernetes.io/evict: true
```
Comment (rmohr) [inline docs/operations/node_assignment.md:152]: This annotation would not be passed on to the pod (well maybe in CNV 1.4 where there was a bug around that, but not no for a very long time).

File: docs/operations/node_assignment.md [hunk 5]
```diff
@@ -140,3 +140,24 @@ An example for `tolerations` may look like this:
...
+policies. Kubevirt VMs are handled as pods with local storage, so by default, 
+descheduler will not evict them. But it can be easily overridden by adding special 
+annotation to the VMI template in the VM:
+
+    ```console
+    spec:
+      template:
+        metadata:
+          annotations:
+            descheduler.alpha.kubernetes.io/evict: true
```
Comment (rmohr) [inline docs/operations/node_assignment.md:158]: @davidvossel what do you  think, if there is no `emptyDir` added as a "disk" in virt-controller, we may actually deploy this annotation by default in our code base.

**Review by rmohr — APPROVED** [review]
/lgtm

File: docs/operations/node_assignment.md (hunk 5 above)
Comment (davidvossel) [reply docs/operations/node_assignment.md:158]: @rmohr i think the annotation should be automated... but i think the situation it should be automatically added is when evictionStrategy: LiveMigrate is set and the VMI is indeed live migratable.

**Comment by ksimon1** [comment]
@codificat can this PR be merged, or should I update somthing?

File: docs/operations/node_assignment.md (hunk 5 above)
Comment (rmohr) [reply docs/operations/node_assignment.md:158]: I am not sure. I was expecting more something like a common baseline with pods. If no emptyDir as workload disk is present, allow the eviction.

Comment (davidvossel) [reply docs/operations/node_assignment.md:158]: i thought we could migrate emptyDir contents. maybe i'm wrong though?

Comment (rmohr) [reply docs/operations/node_assignment.md:158]: Yes, if the VM is migrateable. If it is migrateable, I think we can always add it. I just meant, that I am not sure if we should exclude non-migrateable workloads by default from being recognized by the descheduler.

File: docs/operations/node_assignment.md [hunk 6]
```diff
@@ -140,3 +140,24 @@ An example for `tolerations` may look like this:
...
+## Node balancing with Descheduler
+
+In some cases we might need to rebalance the cluster on current scheduling policy 
+and load conditions. [Descheduler](https://github.com/kubernetes-sigs/descheduler) 
+can find pods, which violates e.g. scheduling decisions and evict them based on descheduler 
+policies. Kubevirt VMs are handled as pods with local storage, so by default, 
+descheduler will not evict them. But it can be easily overridden by adding special 
+annotation to the VMI template in the VM:
+
+    ```console
```
Comment (codificat) [inline docs/operations/node_assignment.md:153]: There is a formatting problem here that you can see in the [rendered preview](https://deploy-preview-387--kubevirt-user-guide.netlify.app/operations/node_assignment/#node-balancing-with-descheduler).

Either remove the triple-quote wrapper (consistent with other yaml examples in this file) or remove the indentation of this section.

**Review by codificat — CHANGES_REQUESTED** [review]
> @codificat can this PR be merged, or should I update somthing?

There is a formatting issue with the yaml sample that must be corrected.

Other than that, my understanding is that the conversation around the annotation does not stop this PR for merging, but if you can confirm that (e.g. resolving the conversation) it would be great, thanks.

**Pull request merged**